### PR TITLE
Fix concurrent write in consumer pending map at startup

### PR DESCRIPTION
**Pull request by ColinSullivan1**
This PR fixes a concurrent write to the consumer pending map on startup reported by a user.

```
fatal error: concurrent map iteration and map write
goroutine 1 [running]:
github.com/nats-io/nats-server/v2/server.(*consumerFileStore).Update(0xc000324cc0, 0xc00107c1e0)
    /home/runner/work/nats-server/src/github.com/nats-io/nats-server/server/filestore.go:6249 +0x352
github.com/nats-io/nats-server/v2/server.(*consumer).writeStoreStateUnlocked(0xc001078900)
    /home/runner/work/nats-server/src/github.com/nats-io/nats-server/server/consumer.go:2133 +0xfb
github.com/nats-io/nats-server/v2/server.(*consumer).checkRedelivered(0xc001078900, 0x0)
    /home/runner/work/nats-server/src/github.com/nats-io/nats-server/server/consumer.go:2047 +0x173
github.com/nats-io/nats-server/v2/server.(*consumer).readStoredState(0xc001078900, 0xc000fb5da0?)
    /home/runner/work/nats-server/src/github.com/nats-io/nats-server/server/consumer.go:2061 +0x85
github.com/nats-io/nats-server/v2/server.(*stream).addConsumerWithAssignment(0xc0003a8300, 0xc0002b8de8, {0x0, 0x0}, 0x0, 0x0?)
```

/cc @nats-io/core

### CODE REVIEW

File: server/filestore.go [hunk 1]
```diff
@@ -6241,6 +6241,10 @@ func (o *consumerFileStore) Update(state *ConsumerState) error {
 		return fmt.Errorf("bad ack floor for stream")
 	}
 
+	// Replace our state.
+	o.mu.Lock()
```
Comment (kozlovic) [inline server/filestore.go:6245]: I mentioned that you should have move that at top of the function, that will also protect access to state.AckFloor, etc...

**Review by kozlovic — APPROVED** [review]
LGTM

**Comment by derekcollison** [comment]
Can we squash down to 1 commit?

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**